src/utils/FileEmitter.ts

```diff
@@ -367,6 +404,49 @@ export class FileEmitter {
 		return this.assignReferenceId(consumedChunk, emittedChunk.id);
 	}
 
+	private emitPrebuiltChunk(
+		emitPrebuiltChunk: Omit<EmittedFile, 'fileName' | 'name'> &
+			Pick<EmittedPrebuiltChunk, 'exports' | 'map'>
+	): string {
+		if (typeof emitPrebuiltChunk.code !== 'string') {
```

Comment:
Both validations should be covered in separate `function` tests, cf. e.g. `test/function/samples/emit-file/invalid-asset-name` for where to place them and what they could look like.